### PR TITLE
Ignore missing qcode in 0.0.1 payloads

### PR DESCRIPTION
**Pull request by kierenbeckett**
### What is the context of this PR?
Fix for bug introduced in #1634, if an answer doesn't have a qcode just omit it from the data in 0.0.1 payloads

### How to review 
Run the 1_0112.json survey and check it gets to the end

### CODE REVIEW

**Comment by shahi645** [comment]
@kierenbeckett linting error

File: tests/app/submitter/test_convert_payload_0_0_1.py [hunk 1]
```diff
@@ -235,6 +235,45 @@ def test_answer_with_multiple_instances(self):
             # Check the converter correctly
             self.assertEqual(answer_object['data']['003'], ['0', '1', '2'])
 
+    def test_answer_without_qcode(self):
```
Comment (shahi645) [inline tests/app/submitter/test_convert_payload_0_0_1.py:144]: could add `with_data_version_0.0.1`

**Comment by shahi645** [comment]
Look good to me, just linting error to sort

**Review by shahi645 — APPROVED** [review]
Kieren to address linting, but otherwise LGTM

File: tests/app/submitter/test_convert_payload_0_0_1.py [hunk 2]
```diff
@@ -235,6 +237,45 @@ def test_answer_with_multiple_instances(self):
...
+                            }
+                        ]
+                    }
+                ]
+            }
+
+            routing_path = [Location(group_id='group-1', group_instance=0, block_id='block-1')]
+
+            convert_answers(self.metadata, QuestionnaireSchema(questionnaire), AnswerStore(user_answer), routing_path)
+
```
Comment (MebinAbraham) [inline tests/app/submitter/test_convert_payload_0_0_1.py]: Maybe worth asserting length of `data` to verify that it got ignored?. Current check will verify that we didn't run into a `KeyError` only?

Comment (MebinAbraham) [inline tests/app/submitter/test_convert_payload_0_0_1.py]: Maybe worth asserting length of data to verify that it got ignored?. Current check will verify that we didn't run into a KeyError only?

**Review by MebinAbraham — COMMENTED** [review]
👇

File: tests/app/submitter/test_convert_payload_0_0_1.py (hunk 2 above)
Comment (MebinAbraham) [inline tests/app/submitter/test_convert_payload_0_0_1.py]: Maybe worth asserting length of `data` to verify that it got ignored?. Current check will verify that we didn't run into a `KeyError` only?

**Review by MebinAbraham — APPROVED** [review]
Happy with the changes, working as expected. 👍 
Worth refactoring the JSON in `0.0.2` to keep it consistent.

**Pull request merged**